src/db/mod.rs

```diff
@@ -48,9 +49,13 @@ impl<T> Repo<T>
 where
     T: diesel::Connection + 'static,
 {
-    /// Constructs a `Repo<T>` for the given database URL (creates a connection pool).
-    pub fn new(database_url: &str) -> Self {
-        Self::from_pool_builder(database_url, r2d2::Builder::default())
+    /// Constructs a `Repo<T>` for the given database config (creates a connection pool).
+    pub fn new(database_config: &DatabaseConfig) -> Self {
+        let mut builder = r2d2::Builder::default();
+        if let Some(max_size) = database_config.connection_pool_max_size {
```

Comment:
Just to reflect the suggested change in `src/config.rs`.
```suggestion
        if let Some(max_size) = database_config.max_conns {
```